scripts/seedUser.js

```diff
@@ -0,0 +1,88 @@
+/* eslint-disable no-console */
+const fs = require('fs');
+const csvParser = require('csv-parser');
+const { Client } = require('pg');
+const moment = require('moment');
+
+
+console.log('Getting postgres URL');
+const env = process.env.NODE_ENV || 'development';
+// eslint-disable-next-line import/no-dynamic-require
+const config = require(`${__dirname}/../config/config.js`)[env];
+
+if (!config.url) {
+  console.error('Postgres url not found. Please ensure you set one in your env.');
+  process.exit(1);
+}
+
+const filePath = process.argv[2];
+
+const client = new Client({
+  connectionString: config.url,
+});
+
+function seedData(data) {
+  // eslint-disable-next-line no-restricted-syntax
+  for (const [index, item] of data.entries()) {
+    const id = parseInt(item.membershipNo, 10);
+    const date = moment().format('YYYY-MM-DD HH:mm');
+    client.query(
+      `
+        DO $$
+        BEGIN
+          IF NOT EXISTS ( SELECT 1 FROM "Users" WHERE "membershipNo" = ${id} )
+          THEN
+            INSERT INTO "Users" ("membershipNo", "firstName", "middleName", "lastName", "password", "createdAt", "updatedAt") VALUES (${id}, '${item.firstName}', '${item.middleName}', '${item.lastName}', 'random', '${date}', '${date}');
+          END IF;
+        END $$;
+        `,
+    ).then(() => {
+      if (index === data.length - 1) {
+        client.end();
+        console.log('User data seeded successfully!');
+      }
+    }).catch((e) => {
+      console.log('An error occured', e.message);
+      if (index === data.length - 1) {
+        client.end();
+      }
+    });
+  }
+}
+
+
+function readFile() {
+  const arr = [];
+
+  fs.createReadStream(filePath)
+    .pipe(csvParser())
+    .on('data', (row) => {
+      arr.push(row);
+    })
+    .on('end', () => {
+      console.log('Seeding data to the database...');
+      seedData(arr);
+    });
+}
+
+function connect() {
```

Comment:
This function doesn't seed the data. It only connects to the database and calls another function that reads the csv file. I already have a function named `seedData`

Reply:
per your statement, the function does more than connect to the database, in essence, it is the main function, and `connect` intuitively says, it does just connects, but it performs all the operations needed to seed the data

same with readFile, it does more than just read file, it reads the file and calls the function that runs the query.

my suggestion is to rename this to maybe `main`, my suggestion is to achieve this

```
async function main() {
  try {
    await client.connect();
    const data = await readFile(filePath);
    await seedData(data);
    console.log('User data seeded successfully!');
  } catch(e) {
     console.log('An error occured', e.message);
  }
  client.end();
}
```

if we use the sequelize seeder file, we can do

```
up: (queryInterface) => { 
    return parseCSVfile(filePath)
                  .then(data => queryInterface.bulkInsert('Users', data));
}
```

Reply:
Your suggestion is great and would have worked if I'm not using es5 for the script. I did this to make it easy to run the script and not have to transpile the file using babel.

Reply:
I followed like a callback approach. Not neat, but it will get the job done.

Reply:
the idea is basically the same, and you are using Promises

instead of async/await, you can use new Promise(resolve, reject)

also, which node version are we using? async/await is available from node8